health-services/project-factory/src/server/validators/campaignValidators.ts

```
@@ -588,28 +650,42 @@
-        return _.isEqual(filteredObj1, filteredObj2);
-    });
-    const duplicateBoundaryRows = boundaryData.filter(e => !uniqueRows.includes(e));
-    const duplicateRowNumbers = duplicateBoundaryRows.map(obj => obj['!row#number!']);
-    const rowNumbersSeparatedWithCommas = duplicateRowNumbers.join(', ');
+    );
+    const seen = new Set<string>();
+    const duplicateRowNumbers: string[] = [];
+    for (const row of boundaryData) {
+        const rowNumber = row["!row#number!"];
+        const rowCopy = { ...row };
+        delete rowCopy["!row#number!"];
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)